frontend/src/HomePage/Folders.jsx

```diff
@@ -317,6 +326,7 @@ export const Folders = function Folders({
               placeholder={t('homePage.foldersSection.folderName', 'folder name')}
               disabled={isCreating || isUpdating}
               value={newFolderName}
+              onKeyDown={(e) => e.key === 'Enter' && handleOnKeySubmit()}
```

Comment:
Simple function execution isn't costly. You can simply move the condition check inside this function.